### ISSUE TITLE
Updates needed to files collected by xcatsnap

### ISSUE THREAD

**Issue by besawn** [xcat2/xcat-core#6942]
While reviewing a recent xcatsnap, I noticed that there are several additional files that would be useful to collect when debugging an issue. Missing files are due to a combination of OS changes and xCAT changes.

On Linux, xcatsnap attempts to collect:
```
"/etc/xcat/*",
"$::ROOTHOME/.xcat/*", 
"$INSTALLDIR/autoinst/*",
"$INSTALLDIR/postscripts/*", 
"$INSTALLDIR/prescripts/*", 
"$INSTALLDIR/custom/*",
"/tftpboot/*",               
"/var/log/consoles/*",
"/etc/*-release",            
"/etc/dhcpd.conf",
"/etc/rsyslog.conf",            
"/etc/rsyslog.d/*",
"/var/lib/dhcpd/dhcpd.leases",
"/etc/hosts",
"/etc/resolv.conf",
"/etc/named.conf",
"/etc/conserver.cf",
"/var/log/conserver",
"/etc/nsswitch.conf", 
"/var/log/messages", 
"/var/log/xcat/*"
```

Some of those paths are not accurate on current OS and xCAT versions. We should keep the existing list so we can continue to collect data on older systems, but add additional data to the set that we collect. After briefly looking through the list, I think we should add:
```
"/etc/dhcp/*",
"/etc/goconserver/server.conf",
"/var/log/goconserver/*"
```

Also, as part of this issue, perform a general review of the data we collect to see if anything else should be added.

**Comment by peterwywong** [comment]
I took a look at RHEL 8.3 (**R**), SLES 12.4 (**S**) and Ubuntu 18.04.2 (**U**). We need to add the following files to be tarred up.

- /etc/dhcp/dpcpd.conf (R, U)
- /var/lib/dhcp/db/dhcpd.leases (S), /var/lib/dhcp/dhcpd.leases (U)
- /etc/goconserver/server.conf, /etc/logrotate.d/goconserver, /var/log/goconserver/* (R, S, U)
- /etc/bind/named.conf (U)

xcatsnap has two code segments in **snap_it** to get the install dir (/install):

    181     chop($INSTALLDIR =
    182           `tabdump site | grep installdir | cut -f2 -d ,`);
    183     $INSTALLDIR =~ s/"//g;

    224     chop($installdir =
    225           `tabdump site | grep installdir | cut -f2 -d ,`);

I don't think the second one is needed. The use of $installdir needs to be adjusted accordingly.

Taking a brief look at **Gathering the system configuration**. 
I think "reventlog -a", "find /tftpboot size -32k", "ls -lR $xcatroot", "lsxcatd -a" fit better in the section of **Capturing xCAT specific information.**

**Comment by peterwywong** [comment]
In **sub run_cmd** of **xcatsnap**, we need to replace

    @output = `$Command`;  **(Line 106)**
by
    @output = `/bin/bash -c \'$Command\'`;

For some distros, the existing command does not work for **ulimit** which is NOT a binary file itself, but part of the shell.

It works fine on RHEL, but not on SLES nor Ubuntu. We find that /tmp/xcatsnap/commands_output/ulimit_a.out is **empty** even though **ulimit -a** at the prompt gives out the expected output.
- - - - -
Currently, /sbin/lspci is used to get lspci data. However, the locations of lspci are different on different distros.
- RHEL 8.3 -- /usr/sbin/lspci
- SLES 12.4 -- /sbin/lspci
- Ubuntu 18.04.2 -- /usr/bin/lspci

We should simply use lspci which is found by the OS.

Moreover, pciutils RPM does not seem to be installed on the RHEL regression nodes, so we need to make sure this is installed on RHEL.
- - - - -
Currently, we have "rpm -qa" to get packages, but Ubuntu needs "apt list". Will add it.
- - - - -
Currently, we have "lssrc -a" where src stands for system resource controller. I do not find it in three nodes of RHEL, SLES and Ubuntu. It is not clear which package has lssrc.

Any suggestion?

**Comment by besawn** [comment]
> Currently, we have "lssrc -a" where src stands for system resource controller. I do not find it in three nodes of RHEL, SLES and Ubuntu. It is not clear which package has lssrc.

`lssrc` is an AIX command, I don't think there is a Linux version available. For commands like `lssrc` and `lspci`, `xcatsnap` should try to collect the data if the command is available, but not require the command by pulling in additional dependencies. If a command is not available, we should continue with the snap.

**Comment by peterwywong** [comment]
"rinv all all" requires the SN and CN being accessible. Otherwise, it seems to get stuck. I waited for a while and then killed it.

"rvitals all all" does not work for VMs, as Mark said, but it comes back with an error message quickly.

For a RHEL 8.3 cluster and a Ubuntu 18.04.2 cluster (SN and CN being accessible), they take **less than 2 minutes** to finish running xcatsnap.

For SLES 15.1 whose SN and CN NOT accessible after regression testing, without including "rinv all all", it takes **less than 1 minute** to run xcatsnap.

**Comment by peterwywong** [comment]
The corresponding PR is https://github.com/xcat2/xcat-core/pull/6952 .

**Comment by besawn** [comment]
Fixed via #6952 .
